### PR TITLE
change: move etcd conf under deployment

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/cli/snippet.lua [hunk 1]
```diff
@@ -70,6 +70,15 @@ function _M.generate_conf_server(env, conf)
         end
     end
 
+    local ssl_trusted_certificate
+    local etcd_tls_verify = etcd.tls.verify
+    if enable_https and etcd_tls_verify then
+        if not conf.apisix.ssl.ssl_trusted_certificate then
+            return nil, "should set ssl_trusted_certificate if etcd tls verify is enabled"
```
Comment (spacewander) [inline apisix/cli/snippet.lua:77]: Now we can remove:
https://github.com/apache/apisix/blob/c397ede02d0b304b8b3a1d5049f6c1d8a83d3804/apisix/cli/snippet.lua#L42
?

File: t/core/etcd-mtls.t [hunk 2]
```diff
@@ -24,6 +24,12 @@ if ($out !~ m/function:/) {
     plan('no_plan');
 }
 
+Test::Nginx::Socket::set_http_config_filter(sub {
```
Comment (spacewander) [inline t/core/etcd-mtls.t:27]: Do we need a separate generator for this test file?

File: apisix/cli/snippet.lua [hunk 3]
```diff
@@ -70,6 +70,15 @@ function _M.generate_conf_server(env, conf)
         end
     end
 
+    local ssl_trusted_certificate
+    local etcd_tls_verify = etcd.tls.verify
```
Comment (spacewander) [inline apisix/cli/snippet.lua:74]: Should we check if etcd.tls is nil?

File: t/core/etcd-mtls.t (hunk 2 above)
Comment (tzssangglass) [reply t/core/etcd-mtls.t:27]: Currently gen_snippet.lua is working fine, except for a problem at https://github.com/apache/apisix/blob/c397ede02d0b304b8b3a1d5049f6c1d8a83d3804/t/core/config.t#L306-L325, but it does not affect the test results.

File: apisix/cli/snippet.lua (hunk 3 above)
Comment (tzssangglass) [reply apisix/cli/snippet.lua]: refactor

File: apisix/cli/snippet.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/cli/snippet.lua]: removed

File: t/core/etcd-mtls.t (hunk 2 above)
Comment (spacewander) [reply t/core/etcd-mtls.t:27]: So can we remove `set_http_config_filter` from this file and use the one set in the t/APISIX.pm?

**Review by spacewander — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**